src/dsp_tools/xmllib/models/dsp_base_resources.py

```diff
@@ -213,13 +199,97 @@ def create_new(
+            color: A hexadecimal color value,
+                [see documentation for details](https://docs.dasch.swiss/latest/DSP-TOOLS/file-formats/xml-data-file/#color).
+                The default value was chosen as it is distinguishable for most color-blind people.
+            status: `active` or `deleted`
+            type_: `circle`, `rectangle` or `polygon` (only the rectangle can be displayed in DSP-APP.
+                The others can be looked at in another frontend, e.g. in TANGOH.)
+
+        Returns:
+            A region resource with the customised shape.
+        """
+        self.geometry = self.geometry.customise_shape(line_width, color, status, type_)
+        return self
+
+    def add_shape_point(self, x: float, y: float) -> RegionResource:
+        """
+        Adds a point to the region shape.
+
+        Please note that the point must be a float between 0 and 1.
+        They represent the location in the image as a percentage of the whole image.
+
+        [See the XML documentation for a visual example.](https://docs.dasch.swiss/latest/DSP-TOOLS/file-formats/xml-data-file/#geometry)
+
+        Args:
+            x: point on the x-axis (width)
+            y: point on the y-axis (height)
+
+        Returns:
+            A region resource with the added point.
+        """
+        self.geometry.points.append(GeometryPoint(x, y, self.res_id))
+        return self
+
+    def add_shape_point_multiple(self, points: list[tuple[float, float]]) -> RegionResource:
+        """
+        Adds multiple points to the region shape.
+
+        Please note that the points must be a float between 0 and 1.
+        They represent the location in the image as a percentage of the whole image.
+
+        [See the XML documentation for a visual example.](https://docs.dasch.swiss/latest/DSP-TOOLS/file-formats/xml-data-file/#geometry)
+
+        Args:
+            points: a list of tuples in the format `[ (x, y) ]`
+
+        Returns:
+            A region resource with the added points.
+        """
+        transformed_points = [GeometryPoint(val[0], val[1], self.res_id) for val in points]
+        self.geometry.points.extend(transformed_points)
+        return self
+
+    def add_shape_point_optional(self, x: Any, y: Any) -> RegionResource:
+        """
+        Adds a point to the region shape.
+
+        Please note that the point must be a float between 0 and 1.
+        They represent the location in the image as a percentage of the whole image.
+
+        [See the XML documentation for a visual example.](https://docs.dasch.swiss/latest/DSP-TOOLS/file-formats/xml-data-file/#geometry)
+
+        A point will only be added if **both x and y** are valid floats.
+
+        Args:
+            x: point on the x-axis (width)
+            y: point on the y-axis (height)
+
+        Returns:
+            A region resource with the added point if both are valid floats.
+        """
+        if any([is_decimal(x), is_decimal(y)]):
```

Comment:
I discussed this with Noemi, we get a warning in any case if a point is not valid, but if one of them is, it can be assumed that the second one is an error in the data, not that it should not be created. So the user should be informed. Also discussed with Noemi, she prefers that these details are not mentioned in the docstring